core/domain/classifier_services.py

```diff
@@ -424,41 +375,209 @@ def delete_classifier_training_job(job_id):
-    """Gets the classifier training job object from the exploration attributes.
+def get_classifier_training_job(exp_id, exp_version, state_name, algorithm_id):
+    """Gets classifier training job object for given algorithm_id for the
+    given <exploration, version, state> triplet.
 
     Args:
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version.
-        state_names: list(str). The state names for which we retrieve the job.
+        state_name: str. The state name for which we retrieve the job.
+        algorithm_id: str. The ID of the algorithm for which classifier training
+            job is to be retrieved.
 
     Returns:
-        list(ClassifierTrainingJob). Domain objects for the Classifier training
-        job model.
+        ClassifierTrainingJob|None. An instance for the classifier training job
+        or None if no such instance is found.
     """
-    training_job_exploration_mapping_models = (
-        classifier_models.TrainingJobExplorationMappingModel.get_models(
-            exp_id, exp_version, state_names))
-    job_ids = []
-    for mapping_model in training_job_exploration_mapping_models:
-        if mapping_model is None:
-            continue
-        job_ids.append(mapping_model.job_id)
+    state_training_jobs_mapping_model = (
+        classifier_models.StateTrainingJobsMappingModel.get_model(
+            exp_id, exp_version, state_name))
+    if state_training_jobs_mapping_model is None:
+        return None
+    job_id = state_training_jobs_mapping_model.algorithm_ids_to_job_ids[
+        algorithm_id]
+    return get_classifier_training_job_by_id(job_id)
 
-    classifier_training_job_models = (
-        classifier_models.ClassifierTrainingJobModel.get_multi(job_ids))
 
-    classifier_training_jobs = []
-    for job_model in classifier_training_job_models:
-        classifier_training_jobs.append(get_classifier_training_job_from_model(
-            job_model))
+def get_state_training_jobs_mapping(exp_id, exp_version, state_name):
+    """Gets training job exploration mapping model for given exploration state
+    combination.
 
-    # Backfill None's to maintain indexes.
-    for index, mapping_model in enumerate(
-            training_job_exploration_mapping_models):
-        if mapping_model is None:
-            classifier_training_jobs.insert(index, None)
-    return classifier_training_jobs
+    Args:
+        exp_id: str. ID of the exploration.
+        exp_version: int. Version of the exploration.
+        state_name: str. Name of the state for which training job mapping model
+            is to be retrieved.
+
+    Returns:
+        StateTrainingJobsMapping. A domain object containing exploration
+        mapping model information.
+    """
+    state_training_jobs_mapping_model = (
+        classifier_models.StateTrainingJobsMappingModel.get_model(
+            exp_id, exp_version, state_name))
+    if state_training_jobs_mapping_model:
```

Comment:
Should we also return in the else case? [Like the way we have done in the `get_classifier_training_job` method?]

Reply:
Done